#755 - Migrate to Spring Boot 4.0.0

michael-82 opened the pull request:
- update spring boot to 4.0.0
- replace undertow with tomcat

michael-82 committed:
#755 - Migrate to Spring Boot 4.0.0

- update spring boot
- use tomcat instead of (removed) undertow
- update to jackson 3

EmteZogaf commented on src/test/java/de/medizininformatikinitiative/dataportal/backend/query/api/status/ValidationIssueSerializerTest.java:
```diff
@@ -0,0 +1,26 @@
+
+  @ParameterizedTest
+  @EnumSource(ValidationIssueType.class)
+  void testSerialize_prefixesCodeAndWritesDetail(ValidationIssueType validationIssueType) {
+    var json = objectMapper.writeValueAsString(validationIssueType);
```
No compiler complains

EmteZogaf commented on src/test/java/de/medizininformatikinitiative/dataportal/backend/query/api/status/UpgradeIssueSerializerTest.java:
```diff
@@ -0,0 +1,26 @@
+
+  @ParameterizedTest
+  @EnumSource(UpgradeIssueType.class)
+  void testSerialize_prefixesCodeAndWritesDetail(UpgradeIssueType upgradeIssueType) {
+    var json = objectMapper.writeValueAsString(upgradeIssueType);
```
No compiler complains

michael-82 committed:
#755 - Migrate to Spring Boot 4.0.0

- propagate JacksonException message instead of throwing blank DataqueryException in DataqueryHandler

EmteZogaf commented on src/main/java/de/medizininformatikinitiative/dataportal/backend/query/dataquery/DataqueryHandler.java:
```diff
@@ -125,7 +125,7 @@ public List<Dataquery> getDataqueriesByAuthor(String userId, boolean includeTemp
     for (de.medizininformatikinitiative.dataportal.backend.query.persistence.Dataquery dataquery : dataqueries) {
       try {
         ret.add(Dataquery.of(dataquery));
-      } catch (JsonProcessingException e) {
+      } catch (JacksonException e) {
         throw new DataqueryException();
       }
```
fixed

EmteZogaf reviewed (approved):
lgtm
